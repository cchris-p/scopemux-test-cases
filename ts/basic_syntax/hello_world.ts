// hello_world.ts
function hello(name: string): string {
  return `Hello, ${name}!`;
}

console.log(hello("World"));
