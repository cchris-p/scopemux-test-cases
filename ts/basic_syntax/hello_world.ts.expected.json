{
    "language": "TypeScript",
    "ast": {
        "type": "translation_unit",
        "functions": [
            {
                "name": "main",
                "parameters": [],
                "body": [
                    {
                        "type": "expression_statement",
                        "expression": {
                            "type": "call_expression",
                            "callee": {
                                "type": "identifier",
                                "name": "console.log"
                            },
                            "arguments": [
                                {
                                    "type": "string_literal",
                                    "value": "Hello, World!"
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}